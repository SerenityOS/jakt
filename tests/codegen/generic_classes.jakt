
class Meep<U, V> {
    public a: U
    public b: V
}

function main(args: [String]) {
    let m = Meep(a: 5, b: 10)
    let n = Meep(a: "Hello!", b: 100)

    let access_m: i32 = m.a
    let access_n: String = n.a
    println("m: {} {}", access_m, m.b)
    println("n: {} {}", access_n, n.b)

    let u: i32 = Meep(a: 5, b: 100).a
}

