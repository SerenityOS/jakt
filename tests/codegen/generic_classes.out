m: 5 10
n: Hello! 100
